src/compiler/checker.ts

```diff
@@ -15711,16 +15717,21 @@ namespace ts {
 
         function checkPropertyAccessExpressionOrQualifiedName(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, right: Identifier) {
             let propType: Type;
-            let leftSymbol = getNodeLinks(left) && getNodeLinks(left).resolvedSymbol;
-            const leftWasReferenced = leftSymbol && getSymbolLinks(leftSymbol).referenced;
             const leftType = checkNonNullExpression(left);
+            const parentSymbol = getNodeLinks(left).resolvedSymbol;
             const apparentType = getApparentType(getWidenedType(leftType));
             if (isTypeAny(apparentType) || apparentType === silentNeverType) {
+                if (isIdentifier(left) && parentSymbol) {
+                    markAliasReferenced(parentSymbol, node);
+                }
                 return apparentType;
             }
             const assignmentKind = getAssignmentTargetKind(node);
             const prop = getPropertyOfType(apparentType, right.escapedText);
             if (!prop) {
+                if (isIdentifier(left) && parentSymbol) {

```

Comment:
Would it read better to lift this check out if the prop/!prop if? Put it afterwards, as:

```ts
if (isIdentifier(left) && parentSymbol && (!prop || !isConstEnumOrConstEnumOnlyModule(prop))) ...
```

(and it would probably read better to invert the !prop/prop if, but that's out of scope here)

Reply:
Probably, considering it's repeated before all exit points of the function.